src/Reflex/TriggerEvent/Base.hs

```diff
@@ -29,21 +33,28 @@ import Reflex.TriggerEvent.Class
 data TriggerInvocation a = TriggerInvocation a (IO ())
 
 -- | A reference to an 'EventTrigger' suitable for firing with 'TriggerEventT'.
-newtype EventTriggerRef t m a = EventTriggerRef { unEventTriggerRef :: Ref m (Maybe (EventTrigger t a)) }
+newtype EventTriggerRef t a = EventTriggerRef { unEventTriggerRef :: IORef (Maybe (EventTrigger t a)) }
```

Comment:
`m` means a lot of different things in a lot of different contexts. In `ImmediateDomBuilderT`, `m` would mean the base monad. But elsewhere, you want to be able to use `DomBuilder t m => ... -> EventTriggerRef t m -> ...` or something. If we want it to be polymorphic, it should just be polymorphic over the `Ref` type, not `m` itself.